### PR TITLE
Extract hydrator logic out of LdapManager. Fix #2

### LINKED ISSUE #2: LdapManager hydrate()
hydrate() method calls two methods, setPassword and setEnabled, they are depending of the user entity implementation .

As workaround Hydrate() method could be customized as explained in the [Cookbook] (https://github.com/Maks3w/FR3DLdapBundle/blob/master/Resources/doc/cookbook/override_ldap-manager.md)

### PR DESCRIPTION
**Pull request by Maks3w**

### CODE REVIEW

**Comment by Maks3w** [comment]
@Noles @bgatellier Would you like to review this proposal for the hydrator process?

I guess later will be added a new method in the HydratorInterface for roles addition.

File: composer.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
     },
     "require-dev": {
         "fabpot/php-cs-fixer": "*",
+        "friendsofsymfony/user-bundle": "~1.3",
```
Comment (Noles) [inline composer.json]: Is this really needed?

File: Hydrator/AbstractHydrator.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+namespace FR3D\LdapBundle\Hydrator;
+
+use FR3D\LdapBundle\Model\LdapUserInterface;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * Provide a hydrator template for easy implementation.
+ */
+abstract class AbstractHydrator implements HydratorInterface
```
Comment (Noles) [inline Hydrator/AbstractHydrator.php]: Must this really be an abstract class?

**Comment by Noles** [comment]
Nice this removes the full dependency to the fos_user_manager. This is very handy.

File: composer.json (hunk 1 above)
Comment (Maks3w) [inline composer.json]: It's the most easy way of mockup a usermanager

File: Hydrator/AbstractHydrator.php (hunk 2 above)
Comment (Maks3w) [inline Hydrator/AbstractHydrator.php]: I wrote this abstract for let the people extend from here and customize the createUser

**Comment by Maks3w** [comment]
Does not remove the dependency yet but is a step more far.

**Pull request merged**